LandLord-core/src/main/java/biz/princeps/landlord/placeholderapi/LLExpansion.java

```diff
@@ -42,54 +37,71 @@ public String onPlaceholderRequest(Player player, String s) {
         if (player == null) {
             return null;
         }
-        int maxClaimPermission = pl.getPlayerManager().getMaxClaimPermission(player);
-        int landcount = wg.getRegionCount(player.getUniqueId());
-        IOwnedLand region = wg.getRegion(player.getLocation());
+        final int maxClaimPermission = pl.getPlayerManager().getMaxClaimPermission(player);
```

Comment:
not quite sure if we shouldn't implement some caching here. This scales very badly if some scoreboard plugin attempts to refresh this value every tick for every player.
Probably a guava cache would help here to keep values for a few seconds.
Probably like this using the key with player as key
https://github.com/eldoriarpg/BloodNight/blob/master/BloodNight-core/src/main/java/de/eldoria/bloodnight/hooks/placeholderapi/Placeholders.java

Reply:
Indeed, thanks! This is a crucial point on which I will act for placeholders involving complex data (database related).